kateryna/databases.py

```diff
@@ -0,0 +1,113 @@
+class PostgreSQLConnectionRefusedError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+
+class Connection(object):
+    def __init__(self, db_name):
+        self.closed = False
+        self.db_name = db_name
+
+    def __str__(self):
+        return "Connection to {}".format(self.db_name)
+
+
+class BaseFramework(object):
+    allowed_commands = ["SELECT", "INSERT", "UPDATE", "DELETE"]
+
+    def __init__(self, connection, **kwargs):
+        self.connection = connection
+        for attr, value in kwargs.items():
+            setattr(self, attr, value)
+
+    def connect(self):
+        self.connection.closed = False
+        return self.connection
+
+    def execute(self, command):
+        if not command.upper() in self.allowed_commands:
+            raise Exception("Provide valid command")
+        return command.upper()
+
+    def close(self):
+        raise NotImplementedError
+
+
+class SQLiteApp(BaseFramework):
+
+    def connect(self):
+        conn = super(SQLiteApp, self).connect()
+        return 'SQLiteApp. {}'.format(conn)
+
+    def execute(self, command):
+        command = super(SQLiteApp, self).execute(command)
+        return "SQLiteApp. {} command has been executed".format(command)
+
+    def close(self):
+        self.connection.closed = True
+        return "{} closed".format(self.connection)
+
+
+class MySQLApp(BaseFramework):
+
+    def connect(self):
+        conn = super(MySQLApp, self).connect()
+        return 'SQLiteApp. {}'.format(conn)
+
+    def execute(self, command):
+        command = super(MySQLApp, self).execute(command)
+        return "MySQLApp. {} command has been executed".format(command)
+
+    def close(self):
+        self.connection.closed = True
+        return "MySQLApp. {} closed".format(self.connection)
+
+
+class PostgreSQLApp(BaseFramework):
+    """"""
+    def __init__(self, credentials, **kwargs):
+        """
+        :param credentials: username and password
+        :type credentials: dict
+        """
+        self.credentials = credentials
+        super(PostgreSQLApp, self).__init__(**kwargs)
+
+    def connect(self):
+        """ if any of credentials is missing, raise an error"""
+        if "username" not in self.credentials or "password" not in self.credentials:
```

Comment:
It's better to fix it, because it easier to review for guys with bad laptops and often it's impossible to find an consensus with 100 symbols in team :)